clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractPartitionAssignor.java

```diff
@@ -34,6 +34,7 @@
  */
 public abstract class AbstractPartitionAssignor implements PartitionAssignor {
     private static final Logger log = LoggerFactory.getLogger(AbstractPartitionAssignor.class);
+    private Map<String, Subscription> subscriptions = null;
```

Comment:
Instead of creating a back channel to propagate this to extensions, maybe we should change the signature of the abstract `assign` method below:
```java
public abstract Map<String, List<TopicPartition>> assign(
  Map<String, Integer> partitionsPerTopic,
  Map<String, Subscription> subscriptions);
```
We can do this in a follow-up if you like.